AcademyChallenge/DataBase/AvatarCoreData.swift

```diff
@@ -106,17 +106,32 @@ class AvatarCoreData {
         //            print(error)
         //            resultHandler(.failure(error))
         //        }
-
     }
+    func delete(avatarObject: Avatar) -> Completable {
+        return Completable.create { completable in
+            let managedContext = self.persistentContainer.viewContext
+            let fetchRequest = NSFetchRequest<NSManagedObject>(entityName: "AvatarEntity")
+            fetchRequest.predicate = NSPredicate(format: "login = %@", avatarObject.login)
 
-//    func delete(avatarObject: Avatar)-> Observable<Avatar> {
-
-//        return Observable<Avatar>.create { observer in
-//            let managedContext = self.persistentContainer.viewContext
-//            let fetchRequest = NSFetchRequest<NSManagedObject>(entityName: "AvatarEntity")
-//            fetchRequest.predicate = NSPredicate(format: "login = %@", avatarObject.login)
-//
-//        }
+            do {
+                let avatarToDelete = try? managedContext.fetch(fetchRequest)
+                if avatarToDelete?.count == 1 {
+                    guard let avatar = avatarToDelete?.first else { return Disposables.create {}}
+                    managedContext.delete(avatar)
+                    try? managedContext.save()
+                }
+            } catch {
+                completable(.error(PersisteError.fetchError))
+                return Disposables.create {}
+            }
+            completable(.completed)
+            return Disposables.create {}
+        }
+    }
+    enum PersisteError: Error {
```

Comment:
Can be moved to its own file

Reply:
I created a file named constantsError and put there